supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystemExpressionFunctions.scala

```diff
@@ -18,7 +19,7 @@ object SharedFileSystemExpressionFunctions {
 }
 
 class SharedFileSystemExpressionFunctions(standardParams: StandardExpressionFunctionsParams)
-  extends StandardExpressionFunctions(standardParams) {
+  extends StandardExpressionFunctions(standardParams) with InputDependentIOFunctionSet {
```

Comment:
I'm not a huge fan of hiding that single (mutable!) field away in a trait...

I think I'd be tempted to make a `val inputSpecificFunctionality = false` a constructor-level member of this class, and also provide a `def forInputs` which provides a copy of this class which can be used to evaluate inputs. something like 

```scala

class SharedFileSystemExpressionFunctions(
  standardParams: StandardExpressionFunctionsParams, 
  forInputs = false) extends ... {
    def makeInputSpecificFunctions = 
      SharedFileSystemExpressionFunctions(standardParams, forInputs = true)
}

```

Reply:
+1